--- a/gypfiles/standalone.gypi
+++ b/gypfiles/standalone.gypi
@@ -72,7 +72,7 @@
           # depending on the packages installed on the local machine. Set this
           # to 0 to build against locally installed headers and libraries (e.g.
           # if packaging for a linux distro)
-          'use_sysroot%': 1,
+          'use_sysroot%': 0,
         },
         'host_arch%': '<(host_arch)',
         'target_arch%': '<(target_arch)',
