--- a/src/third_party/libdrm/src/xf86drm.c
+++ b/src/third_party/libdrm/src/xf86drm.c
@@ -61,6 +61,7 @@
 #include <sys/sysmacros.h>
 #endif
 #include <math.h>
+#include <sys/sysmacros.h>
 
 /* Not all systems have MAP_FAILED defined */
 #ifndef MAP_FAILED
@@ -2765,7 +2766,7 @@ static char *drmGetMinorNameForFD(int fd, int type)
 {
 #ifdef __linux__
     DIR *sysdir;
-    struct dirent *pent, *ent;
+    struct dirent *ent;
     struct stat sbuf;
     const char *name = drmGetMinorName(type);
     int len;
@@ -2797,24 +2798,17 @@ static char *drmGetMinorNameForFD(int fd, int type)
     if (name_max == -1)
         goto out_close_dir;
 
-    pent = malloc(offsetof(struct dirent, d_name) + name_max + 1);
-    if (pent == NULL)
-         goto out_close_dir;
-
-    while (readdir_r(sysdir, pent, &ent) == 0 && ent != NULL) {
+    for (ent = readdir(sysdir); ent; ent = readdir(sysdir)) {
         if (strncmp(ent->d_name, name, len) == 0) {
             snprintf(dev_name, sizeof(dev_name), DRM_DIR_NAME "/%s",
                  ent->d_name);
 
-            free(pent);
             closedir(sysdir);
 
             return strdup(dev_name);
         }
     }
 
-    free(pent);
-
 out_close_dir:
     closedir(sysdir);
 #else
