--- a/script/build-libchromiumcontent.py
+++ b/script/build-libchromiumcontent.py
@@ -42,6 +42,7 @@ def main():
   bootstrap = os.path.join(script_dir, 'bootstrap')
   update = os.path.join(script_dir, 'update')
   build = os.path.join(script_dir, 'build')
+  patch = os.path.join('/', 'usr', 'bin', 'patch')
   create_dist = os.path.join(script_dir, 'create-dist')
   if args.force_update or libchromiumcontent_outdated():
     execute_stdout([sys.executable, bootstrap])
@@ -54,6 +55,13 @@ def main():
                     '--no_zip', '--keep-debug-symbols',
                     '-t', args.target_arch])
   else:
+    execute_stdout([patch, '-Np1', '-i', '<use-system-root-patch>'])
+    execute_stdout([patch, '-Np1', '-i', '<readdir-fix-patch>'])
+    execute_stdout([patch, '-Np1', '-i', '<libdrm-makedev-fix-patch>'])
+    execute_stdout([patch, '-Np1', '-i', '<dma-buf-sync-patch>'])
+    execute_stdout([patch, '-Np1', '-i', '<v8-use-system-mksnapshot>'])
+    execute_stdout([patch, '-Np1', '-i', '<v8-get-array-buffer-allocator>'])
+    execute_stdout([patch, '-Np1', '-i', '<webkit-fix>'])
     execute_stdout([sys.executable, build, '-R', '-t', args.target_arch])
     execute_stdout([sys.executable, create_dist, '-c', 'static_library',
                     '--no_zip', '-t', args.target_arch])
