--- a/src/build/config/sysroot.gni
+++ b/src/build/config/sysroot.gni
@@ -42,7 +42,7 @@ if (current_os == target_os && current_cpu == target_cpu &&
   # By default build against a sysroot image downloaded from Cloud Storage
   # during gclient runhooks.
   if (current_cpu == "x64") {
-    sysroot = "$target_sysroot_dir/debian_jessie_amd64-sysroot"
+    sysroot = "/"
   } else if (current_cpu == "x86") {
     sysroot = "$target_sysroot_dir/debian_jessie_i386-sysroot"
   } else if (current_cpu == "mipsel") {
--- a/src/v8/gypfiles/standalone.gypi
+++ b/src/v8/gypfiles/standalone.gypi
@@ -95,7 +95,7 @@
                 'sysroot%': '<!(cd <(DEPTH) && pwd -P)/build/linux/debian_jessie_arm-sysroot',
               }],
               ['target_arch=="x64"', {
-                'sysroot%': '<!(cd <(DEPTH) && pwd -P)/build/linux/debian_jessie_amd64-sysroot',
+                'sysroot%': '/',
               }],
               ['target_arch=="ia32"', {
                 'sysroot%': '<!(cd <(DEPTH) && pwd -P)/build/linux/debian_jessie_i386-sysroot',
